interview-questions.md

```diff
@@ -8,62 +8,75 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is repeating a process over and over again until a condition is met. Most cases it is used in relation to iterating through strings or arrays.
 
-  Researched answer:
+  Researched answer: In reference to code, iteration refers to the execution of a block of code repeatedly until a condition is met (performing repetitive tasks). Specifically, we use iteration for any data types with the length property meaning strings and arrays. An example of iteration would be 'for' loops. 
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: .map() takes in value(element), index, and array. Value is required, but index and array are optional.
 
-  Researched answer:
+  Researched answer:.map() is a high order function that iterates through an array and will return an array with the same length. .map() has a required parameter called callbackFn which refers to the function that mutates every element of the argument array and returns the result into a new array. It also requires the parameter element (or value) which refers to the element processed as it iterates through the argument array. There are also 3 other optional parameters: index which refers to the location of the current element as it iterates through the array, array which is the array that .map(). was called on, and thisArg which is used to specify a different argument than what 'this' may actually be referencing(?).
+  
+  //Hey Austin, would you let me know if im understanding thisArg correctly? (if my understanding is wrong, would you let me know what it actually refers to?)
+  Thanks!
```

Comment:
Yea, So I have only used it ("this") once in a map. 

I passed it to the argument after all of the map syntax was completed and logic was built out. 
```javascript
obj.map((value)=>{
 let someLogic = typeof value == 'number' ? this.objKey * value : 0
 return someLogic
},this)